src/adapters/poi/idunn_poi.js

```diff
@@ -81,28 +81,35 @@ export default class IdunnPoi extends Poi {
     }
   }
 
-  static async poiApiLoad(id, options = {}) {
+  static async poiApiLoad(obj, options = {}) {
     let rawPoi = null;
-    const url = `${serviceConfig.idunn.url}/v1/places/${id}`;
+    const url = `${serviceConfig.idunn.url}/v1/places/${obj.id}`;
     let requestParams = {};
     if (options.simple) {
       requestParams = {verbosity: 'short'};
     }
     try {
-      rawPoi = await Ajax.getLang(url, requestParams);
+      let headers = {};
+      if (obj.suggestionRank !== undefined && obj.query !== undefined) {
+        if (obj.position !== undefined) {
+          const pos = obj.position;
+          headers['X-QwantMaps-FocusPosition'] = `${pos.lat};${pos.lon};${pos.zoom}`;
+        }
+        headers['X-QwantMaps-Query'] = encodeURI(obj.query);
+        headers['X-QwantMaps-SuggestionRank'] = obj.suggestionRank;
+      }
+      rawPoi = await Ajax.getLang(url, requestParams, {}, headers);
       return new IdunnPoi(rawPoi);
     } catch (err) {
-      if (err === 404) {
-        return;
-      } else {
+      if (err !== 404) {
         const s_requestParams = JSON.stringify(requestParams);
         Error.sendOnce(
           'idunn_poi', 'poiApiLoad',
           `unknown error getting idunn poi reaching ${url} with options ${s_requestParams}`,
           err
         );
-        return;
       }
+      return;
```

Comment:
This `return` is useless here.
Another way would be to do a real early return by keeping the `err === 404` test but ditching the `else`.
IMO it's a better pattern to manage the easy cases first.